packages/adders/paraglide/index.ts

```diff
@@ -66,21 +50,29 @@ export default defineAdder({
 	environments: { svelte: false, kit: true },
 	homepage: 'https://inlang.com',
 	options,
-	packages: [
-		{
-			name: '@inlang/paraglide-sveltekit',
-			version: '^0.11.1',
-			dev: false
-		}
-	],
-	files: [
-		{
-			// create an inlang project if it doesn't exist yet
-			name: () => 'project.inlang/settings.json',
-			condition: ({ cwd }) => !fs.existsSync(path.join(cwd, 'project.inlang/settings.json')),
-			content: ({ options, content }) => {
+	run: ({ api, cwd, options, typescript, kit, dependencyVersion }) => {
```

Comment:
```suggestion
	run: ({ files, packages, scripts, cwd, options, typescript, kit, dependencyVersion }) => {
```
Open for discussion: Only one export, or multiple? I had implemented both locally, but decided to go with `api` as `files` and `scripts` only had one property. It might be more future proof, if we export all three things separately.

Reply:
i would probably name them as just `sv` and `file`. `sv` would encapsulate `packages` and `scripts` and `file` would just be for file handling. 

but maybe having a unified `sv` that does it all isn't so bad... hmmm, need to think on it.

Reply:
interesting approach to just name it `sv`, i like that. 
Although i wouldn't be sure about combining `packages` and `scripts` and leaving `files` separately. I would suggest that we do all or nothing here.

Reply:
the issue with not letting it have it's own namespace is that each method now gets a bit more verbose

ex:
```js
sv.updateFile
sv.createOrUpdateFile
sv.createFile
```
as opposed to:
```js
file.update
file.createOrUpdate
sv.create
```

Reply:
But wouldn't we have exactly the same problem with `script` and `package`?
Additionally, `package` is a reserved keyword in strict mode, which would make it necessary to use the plural or something completely different like `dependencies`

Reply:
Another option would be to allow `sv.file.update()` to group it even more. But in that case i would probably start destructuring the object, basically leaving us with an option we already have.

Reply:
`pkg` or `pkgJson` or something would be another option

I haven't made up my mind about splitting vs combining, but do feel we should be consistent. maybe we should ask Rich for feedback after svelte.dev is updated as he's quite good with API design

Reply:
thinking on it more, I think a single api via `sv` should be fine:
```ts
sv.file // creates or updates files - see other comment down below
sv.dependency // adds dep
sv.devDependency // adds devDep
sv.execute // runs any scripts
sv.dependsOn // runs another adder(?)
```

Reply:
Generally, I agree.

The problem I have is with the `dependsOn` as described somewhere else.
We need to know the adders something depends on before running the `run` method, so that we can ask the appropriate questions

After some rapid prototyping I came up with something like this:
```js
// we need to collect all todos to detect the `dependsOn` dependencies
// so that we can ask the relevant questions beforehand
const todos: Array<{
	type: 'file' | 'dependency' | 'script' | 'dependsOn';
	data: SvApi['file'] | SvApi['execute'] | SvApi['dependsOn'] | Dependency;
}> = [];
```
The problem with this is, that it's still pretty basic, and I would still need to store the adder information and the (potentially) workspace somewhere. After collecting them for all adders we could filter out the `type: 'dependsOn'` ones and ask the required questions. Then we could run the `todos` in the same order we collected them to preserver consistency.

There are two alternatives i see:
* The adder that is being applied (`lucia` in our case) is responsible for asking the relevant `drizzle` questions, based on some sort of condition
* We make the current `dependsOn` array a function, and remove it from the sv api, so that we can collect such stuff beforehand.

Reply:
I have just moved `dependsOn` out of the `sv` api, because the problems mentioned above. For now I have decided to make `dependsOn` a function accepting a workspace, so that every add-on is responsible for deciding if it needs to run or not. This works great, and I think this is the best solution we can find at this point.